clients/python/client/osparc/__init__.py

```diff
@@ -34,14 +36,24 @@
     ValidationError,
     __version__,
 )
+from packaging.version import Version
 
-from ._exceptions import RequestError
+from ._exceptions import RequestError, VisibleDeprecationWarning
 from ._files_api import FilesApi
 from ._info import openapi
 from ._solvers_api import SolversApi
 from ._studies_api import StudiesApi
 from ._utils import dev_features_enabled
 
+if Version(python_version()) < Version("3.8.0"):
```

Comment:
I think you should do a dinstinction between **deprecation** and **retired** ... (pls keep this workflow around in the doc of the client. It is from https://www.ibm.com/docs/en/acvfc?topic=manager-product-lifecycle)

![Pasted image 20240322113853](https://github.com/ITISFoundation/osparc-simcore-clients/assets/32402063/705d6817-1662-4491-8644-75e1a19ff2eb)

When the user `import osparc` here, it should somehow warn that 3.6 is **deprecated** but if he opens with 3.5 it should say that it is **retired**. 

Somehow you need to define a logic that identifies:
- what is deprecated
- when is going to be retired
- what version is going to be the replacement.


I would prepare this code to support both states. i.e. deprecation, and retired.
```python

PYTHON_VERSION_DEPRECATED = Version("3.6")
PYTHON_VERSION_NEW_VERSION = Version("3.8.0")

current_version = Version(python_version())

if  current_version.base_version == PYTHON_VERSION_DEPRECATED:
    warning_msg: str = (
        "This is the final version of osparc which "
        f"will support Python {current_version}. "
        f"Future versions of osparc will only work with Python version >={PYTHON_VERSION_NEW_VERSION}."
    )
    warnings.warn(warning_msg, VisibleDeprecationWarning)
    
```

Reply:
Good point. I have tried to improve the logic. I think it is important we get this right, so I would be happy if you would have another look. Thanks